report/report.go

```diff
@@ -469,6 +475,37 @@ func (r Report) upgradeNamespaces() Report {
 	return r
 }
 
+// Note in-place modification of Endpoint.Nodes
```

Comment:
This is way dodgy. Don't do it. If you must, I expect an essay here explaining why it is safe.

Reply:
I was under the impression the existing `upgrade` routines did it too, but on closer inspection they don't.  Will fix.